addon/components/frost-modal-binding.js

```diff
@@ -81,13 +143,40 @@ const FrostModalBinding = Component.extend(PropTypesMixin, {
       this.onClose()
     },
 
+    /**
+     * Handle confirmation in a promise-aware way.
+     * If onConfirm returns a promise, closeOnConfirm will wait for it to resolve successfully before closing the modal.
+     * @returns {Object} the return value of onConfirm.
+     * @private
+     */
     _onConfirm () {
       const onConfirm = this.get('onConfirm')
+      let confirmed
       if (onConfirm) {
-        onConfirm()
+        confirmed = onConfirm()
```

Comment:
What happens if this fails?  Line 165 or 170 isn't going to run.

Reply:
If the `onConfirm` promise fails, the dialog will not close, because of the way `.then` chaining works. This is good because it lets us keep the dialog open while an error notification (shown by `onConfirm`) is displayed, allowing the user to fix their input.

> For example, one might try to Create a New User, and get an error back form the API that the email address is already in use, or some other server-knowledge-only error

**Good catch** though about the state of the disabled button not working right, though.  I don't want `onClose` to be invoked, but I still want the confirm button to be re-enabled even if the promise is rejected.  Can I do 

```js
confirmed
  .finally(() => { ... })  // fix button state
  .then(() => {...})  // invoke onClose
```
or do I need to move this block so that the final action (re-enabling confirmation) is done after the `.onClose()` invocation so that I end up with
```js
confirmed
  .then(() => {...})  // invoke onClose
  .finally(() => { ... })  // fix button state
```
?